stripe/src/test/java/com/stripe/android/StripeKtxTest.kt

```diff
@@ -0,0 +1,552 @@
+        crossinline repositoryInvocationBlock: suspend (RepositoryParam, ApiRequest.Options) -> APIObject?,
+        crossinline createAPIInvocationBlock: suspend (String, String?, String?) -> APIObject
+    ): Unit = runBlocking {
+        whenever(
+            repositoryInvocationBlock(any(), any())
+        ).thenThrow(mock<AuthenticationException>())
+
+        assertFailsWith<AuthenticationException> {
+            createAPIInvocationBlock(
+                "Dummy String param",
+                TEST_IDEMPOTENCY_KEY,
+                TEST_STRIPE_ACCOUNT_ID
+            )
+        }
+    }
+
+    private inline fun <APIObject : Any, reified RepositoryParam : Any>
+    `Given repository returns empty value when calling createAPI with String param then throws APIException`(
+        crossinline repositoryInvocationBlock: suspend (RepositoryParam, ApiRequest.Options) -> APIObject?,
+        crossinline createAPIInvocationBlock: suspend (String, String?, String?) -> APIObject
+    ): Unit = runBlocking {
+        whenever(
+            repositoryInvocationBlock(any(), any())
+        ).thenReturn(null)
+
+        assertFailsWith<InvalidRequestException> {
+            createAPIInvocationBlock(
+                "Dummy String param",
+                TEST_IDEMPOTENCY_KEY,
+                TEST_STRIPE_ACCOUNT_ID
+            )
+        }
+    }
+
+    private inline fun <reified APIObject : Any>
+    `Given repository returns non-empty value when calling retrieveAPI with String param then returns correct result`(
+        crossinline repositoryInvocationBlock: suspend (String, ApiRequest.Options, List<String>) -> APIObject?,
+        crossinline retrieveAPIInvocationBlock: suspend (String, String?) -> APIObject
+    ) = runBlocking {
+        val expectedApiObj = mock<APIObject>()
+
+        whenever(
+            repositoryInvocationBlock(any(), any(), any())
+        ).thenReturn(expectedApiObj)
+
+        val actualObj = retrieveAPIInvocationBlock(
+            "Dummy String param",
+            TEST_STRIPE_ACCOUNT_ID
+        )
+
+        assertSame(expectedApiObj, actualObj)
+    }
+
+    private inline fun <APIObject : Any>
+    `Given repository throws exception when calling retrieveAPI with String param then throws same exception`(
+        crossinline repositoryInvocationBlock: suspend (String, ApiRequest.Options, List<String>) -> APIObject?,
+        crossinline retrieveAPIInvocationBlock: suspend (String, String?) -> APIObject
+    ): Unit = runBlocking {
+        whenever(
+            repositoryInvocationBlock(any(), any(), any())
+        ).thenThrow(mock<AuthenticationException>())
+
+        assertFailsWith<AuthenticationException> {
+            retrieveAPIInvocationBlock(
+                "Dummy String param",
+                TEST_STRIPE_ACCOUNT_ID
+            )
+        }
+    }
+
+    private inline fun <APIObject : Any>
```

Comment:
make this `suspend` and remove `runBlocking`

Reply:
this is a template function for the actual tests, we'll need to add runBlocking to the actual tests if this becomes suspend. replaced with `testDispatcher.runBlockingTest`